payroll_hr_public_holidays/models/hr_payslip.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 2021 Nimarosa (Nicolas Rodriguez) (<nicolasrsande@gmail.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime, time
+
+from odoo import _, api, models
+
+
+class HrPayslip(models.Model):
+    _inherit = "hr.payslip"
+
+    @api.model
+    def get_worked_day_lines(self, contracts, date_from, date_to):
+        res = super().get_worked_day_lines(contracts, date_from, date_to)
+        for contract in contracts.filtered(
+            lambda contract: contract.resource_calendar_id
+        ):
+            day_from = datetime.combine(date_from, time.min)
+            day_to = datetime.combine(date_to, time.max)
+            day_contract_start = datetime.combine(contract.date_start, time.min)
```

Comment:
It's a datetime because this fields come from payslip. Look at original function I just replicated what is in hr_payslip to have the exact same values that other worked days functions use. 

I think it's a datetime because I allows to use relativedeltas and that's why payroll module use it. It's makes calculation better. In this case, it's not necessary to use datetimes but its payroll module who is using it.

Reply:
Done.